Core/Utilities/test/Test_DAG.py

```diff
@@ -13,6 +13,17 @@
 
 __RCSID__ = "$Id $"
 
+def listToSet(obj):
```

Comment:
I am not sure this is correct. You may want to have duplicated items in a list. Why wasn't `sorted` good enough ?

Reply:
Because the list contains a mixtures of types which have no sensible comparison operator between them. In Python 2 this was poorly defined and could result in bizarre failures. In [Python 3 this is now an error](https://portingguide.readthedocs.io/en/latest/comparisons.html#unorderable-types):

```python
In [2]: sorted(["a", 1, dict()])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-adfb206d711c> in <module>
----> 1 sorted(["a", 1, dict()])

TypeError: '<' not supported between instances of 'int' and 'str'
```

These lists internally come from `dict` and `set` objects which have no stable ordering in Python 2 (`dicts` are in Python 3.7+ but `set` still isn't and I don't like trusting either).

A lot of these tests that I've fixed were comparing to the "random" order than Python 2.7 happens to produce if executed in this order. A few places were so unstable that introducing `pdb.set_trace()` was enough to trigger them to reshuffle (probably due to the garbage collector having a chance to run).

Reply:
Hum, another "nice" feature of python 3 I guess....